rcl_yaml_param_parser/include/rcl_yaml_param_parser/types.h

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef RCL_YAML_PARAM_PARSER__TYPES_H_
+#define RCL_YAML_PARAM_PARSER__TYPES_H_
+
+#include "rcutils/allocator.h"
+#include "rcutils/types/string_array.h"
+
+/// \typedef bool_array_t
+/// \brief Array of bool values
+typedef struct rcl_bool_array_s
+{
+  bool * values;
+  size_t size;
+} rcl_bool_array_t;
+
+/// \typedef int64_array_t
+/// \brief Array of int64_t values
+typedef struct rcl_int64_array_s
+{
+  int64_t * values;
+  size_t size;
+} rcl_int64_array_t;
+
+/// \typedef double_array_t
+/// \brief Array of double values
+typedef struct rcl_double_array_s
+{
+  double * values;
+  size_t size;
+} rcl_double_array_t;
+
+/// \typedef variant_t
+/// \brief variant_t stores the value of a parameter
+/// Only one pointer in this struct will store the value
+typedef struct rcl_variant_s
+{
+  bool * bool_value;  ///< If bool, gets stored here
+  int64_t * integer_value;  ///< If integer, gets stored here
+  double * double_value;  ///< If double, gets stored here
+  char * string_value;  ///< If string, gets stored here
+  // byte_array_t * byte_array_value;  ///< If array of bytes
```

Comment:
Will this be populated and use in the future?
It should either have a "TODO" explaining what remains to be done or be deleted if this is not planned to be supported

Reply:
Will not support for now so removed it. When we have some time, will discuss and implement it

Reply:
> Will not support for now so removed it. When we have some time, will discuss and implement it

If we agree that we should support for it. The field should be part of the datasrtucture.
It can always be NULL pending implementations but this allow us to have a well defined interface if people want to interact with these structures. Otherwise we may break when we add the field that is undesired in my opinion

Reply:
If you agree with the previous comment it would be great to add the field back